src/storage/src/hummock/sstable/sstable_object_id_manager.rs

```diff
@@ -198,25 +199,57 @@ impl GetObjectId for Arc<SstableObjectIdManager> {
     }
 }
 
+struct SharedComapctorObjectIdManagerCore {
+    output_object_ids: VecDeque<u64>,
+    client: GrpcCompactorProxyClient,
+}
+impl SharedComapctorObjectIdManagerCore {
+    pub fn new(output_object_ids: VecDeque<u64>, client: GrpcCompactorProxyClient) -> Self {
+        Self {
+            output_object_ids,
+            client,
+        }
+    }
+}
 /// `SharedComapctorObjectIdManager` is used to get output sst id for serverless compaction.
 #[derive(Clone)]
 pub struct SharedComapctorObjectIdManager {
-    output_object_ids: VecDeque<u64>,
+    core: Arc<tokio::sync::Mutex<SharedComapctorObjectIdManagerCore>>,
 }
 
 impl SharedComapctorObjectIdManager {
-    pub fn new(output_object_ids: VecDeque<u64>) -> Self {
-        Self { output_object_ids }
+    pub fn new(output_object_ids: VecDeque<u64>, client: GrpcCompactorProxyClient) -> Self {
+        Self {
+            core: Arc::new(tokio::sync::Mutex::new(
+                SharedComapctorObjectIdManagerCore::new(output_object_ids, client),
+            )),
+        }
     }
 }
 
 #[async_trait::async_trait]
 impl GetObjectId for SharedComapctorObjectIdManager {
     async fn get_new_sst_object_id(&mut self) -> HummockResult<HummockSstableObjectId> {
-        if let Some(first_element) = self.output_object_ids.pop_front() {
+        let mut guard = self.core.lock().await;
+        let core = guard.deref_mut();
+
+        if let Some(first_element) = core.output_object_ids.pop_front() {
             Ok(first_element)
         } else {
-            return Err(HummockError::other("Output object id runs out"));
+            tracing::warn!("The pre-allocated object ids are used up, and new object id are obtained through RPC.");
+            let request = GetNewSstIdsRequest { number: 10 };
```

Comment:
I think we can use this config https://github.com/risingwavelabs/risingwave/blob/94c59b8668f266b42593940942fbb64e0379bdb5/src/storage/src/opts.rs#L60 instead of hardcoding the fetch number here. This makes it easier to change if we find this number not reasonable in the future.

Reply:
Agree, fixed.